pkg/reporting/send.go

```diff
@@ -37,8 +41,10 @@ type eventSender[E any] struct {
 }
 
 func (s eventSender[E]) senderLoop(ctx context.Context, logger *zap.Logger) {
-	ticker := time.NewTicker(time.Second * time.Duration(s.client.BaseConfig.PushEverySeconds))
-	defer ticker.Stop()
+	heartbeat := time.Second * time.Duration(s.client.BaseConfig.PushEverySeconds)
```

Comment:
nitpick (out of context): BaseConfig could have those fields defines as time.Duration which supports json marshaling

Reply:
IIUC `time.Duration` doesn't have fancy JSON marshaling (it's just an int64 of nanoseconds), unless I'm missing something?

Reply:
omg, you are right, I messed it with the time.ParseDuration which I always used with custom type implementing Unmarshal method on it.